buildenv/buildenv.nix

```diff
@@ -0,0 +1,278 @@
+{
+  activationScripts_storePath ? "@activationScripts@",
+  defaultEnvrc_storePath ? "@defaultEnvrc@",
+  coreutils_storePath ? "@coreutils@",
+  floxBuildenv_storePath ? "@out@",
+  manifest,
+  name ? "environment",
+  serviceConfigYaml ? "",
+}:
+let
+
+  # Start by converting (text) storePaths into contextual strings.
+  # FTR a contextual string is one that knows it's a derivation!
+  # This is extremely important for populating the Nix dependency
+  # graph. See `builtins.getContext` and `addContextFrom` in nixpkgs
+  # strings.nix.
+  activationScripts = builtins.storePath activationScripts_storePath;
+  defaultEnvrc = builtins.storePath defaultEnvrc_storePath;
+  floxBuildEnv = builtins.storePath floxBuildenv_storePath;
+  coreutils = builtins.storePath coreutils_storePath;
+
+  # A helpful library function copied from nixpkgs/lib/attrsets.nix.
+  # foldlAttrs = f: init: set:
+  #   builtins.foldl'
+  #     (acc: name: f acc name set.${name})
+  #     init
+  #     (builtins.attrNames set);
+
+  # The system we're building for.
+  system = builtins.currentSystem;
+
+  # Copy manifest file into the store for access within derivations.
+  manifestLock = /. + manifest;
```

Comment:
The current version of [src/libexpr/primops.cc](https://github.com/NixOS/nix/blob/a8e600e3864dbc677474c1cd7ed339e36bfa6b93/src/libexpr/primops.cc#L1617-L1618) in github:NixOS/nix suggests:
```
      **DEPRECATED.** Use `/. + "/path"` to convert a string into an absolute
      path. For relative paths, use `./. + "/path"`.
```

Nix documentation is literally everywhere, but I tend to respect comments in source code above all else. Regardless, I'm interpreting this comment as telling me that `builtins.toPath()` is deprecated, and recommending the use of `/. + foo` rather than `builtins.path()`. I don't see a similar deprecation warning for `builtins.path()`, FTR.

Thereafter I guess I don't care - whatever works works. I'm just trying to follow the best advice as found in the source code.

Reply:
It says that about `toPath` correct.
`path` is arguably more powerful as well (e.g. allows renaming the file in store), i dont think there is much of a difference between `/. + foo` and `builtins.path {path = "/foo"; }` other than maybe laziness (probably  not).
visually i prefer `builtins.path` when its also used in a file that mixes it with `builtins.storepath`.

But i did mark it as non-blocking so since it doesn't break anything we can go with it either way.

Reply:
Thanks for the context. As before I don't care, was just trying to follow best practice and maybe over-rotated on the significance of comments in the code. I did as you asked in commit 4ce2e3e649517936108db0ee6fc5fd30a9d69a2c - resolving conversation.